### PR TITLE
User context2

### PR DESCRIPTION
**Pull request by Etroid**
## What, Why?
Updated the functions related to user and roles in Drupal 8.

```
Scenario: User exists and has roles
    When I examine the user "admin"
    Then the user should have the roles "Administrator"
    Then the user should not have the roles "Editor"
```

@damontgomery ^^

### CODE REVIEW

File: src/Palantirnet/PalantirBehatExtension/Context/EntityDataContext.php [hunk 1]
```diff
@@ -232,18 +232,20 @@ public function assertEntityIsUser()
...
      * Verify that a user account has a set of roles.
      *
-     * @param stdclass $account A Drupal user account object.
-     * @param array    $roles   An array of Drupal role objects.
+     * @param \Drupal\Core\Entity\EntityInterface $account
+     *  A Drupal user account object.
+     * @param array $roles
+     *  An array of Drupal role objects.
+     *
+     * @throws \Exception if user does not have one ore more roles as defined in $roles
```
Comment (damontgomery) [inline src/Palantirnet/PalantirBehatExtension/Context/EntityDataContext.php]: There is a typo in `one ore`.

**Review by damontgomery — COMMENTED** [review]
This looks good to me. Thank @Etroid. I haven't had a chance to try it out, but I'll try to tackle that this week.

File: src/Palantirnet/PalantirBehatExtension/Context/EntityDataContext.php [hunk 2]
```diff
@@ -232,18 +232,20 @@ public function assertEntityIsUser()
...
      */
     public function assertUserRoles($account, $roles)
     {
-        throw new NotUpdatedException('Method not yet updated for Drupal 8.');
-
         foreach ($roles as $role) {
-            if (false === user_has_role($role->rid, $account)) {
-                throw new \Exception(sprintf('User "%s" does not have role "%s".', $account->name, $role->name));
+            if (false === $account->hasRole($role->id())) {
+                throw new \Exception(sprintf('User "%s" does not have role "%s".', $account->name(), $role->label()));
```
Comment (damontgomery) [inline src/Palantirnet/PalantirBehatExtension/Context/EntityDataContext.php]: I get an error for this that the function `name()` doesn't exist. I was able to replace it with `$account->getAccountName()` and that worked.

File: src/Palantirnet/PalantirBehatExtension/Context/EntityDataContext.php [hunk 3]
```diff
@@ -253,18 +255,20 @@ public function assertUserRoles($account, $roles)
...
      */
     public function assertNotUserRoles($account, $roles)
     {
-        throw new NotUpdatedException('Method not yet updated for Drupal 8.');
-
         foreach ($roles as $role) {
-            if (true === user_has_role($role->rid, $account)) {
-                throw new \Exception(sprintf('User "%s" has role "%s".', $account->name, $role->name));
+            if (true === $account->hasRole($role->id())) {
+                throw new \Exception(sprintf('User "%s" has role "%s".', $account->name(), $role->label()));
```
Comment (damontgomery) [inline src/Palantirnet/PalantirBehatExtension/Context/EntityDataContext.php]: Same as above. I needed to replace `$account->name()` with `$account->getAccountName()`.

**Review by damontgomery — CHANGES_REQUESTED** [review]
@Etroid, Sorry for the long delay on reviewing this. I tested this out and got errors when the tests failed since the `name()` method did not exist.

Could you try the suggested changes and let me know if that works for you?

**Comment by Etroid** [comment]
@damontgomery thanks for reviewing. You're absolutely right. Using `getAccountName` as found on `\Drupal\User\Entity\User` works like a charm 👍

**Review by damontgomery — APPROVED** [review]

**Comment by damontgomery** [comment]
Thanks @Etroid!

**Pull request merged**